### PR TITLE
feat: add constraint controller

### PR DESCRIPTION
**Pull request by zreigz**

### CODE REVIEW

File: internal/controller/constraint_controller.go [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+)
+
+const missingLabelError = "missing label while attempting to map a constraint status resource"
+
+type ConstraintReconciler struct {
+	client.Client
+	Scheme        *runtime.Scheme
+	ConsoleClient consoleclient.Client
+	Reader        client.Reader
+	Constrains    map[string]*console.PolicyConstraintAttributes
```
Comment (michaeljguarino) [inline internal/controller/constraint_controller.go:32]: I think this should be spelled `Constraints` everywhere

**Review by michaeljguarino — COMMENTED** [review]
Is there a constraint pod status object for every instance of a constraint template?

**Comment by zreigz** [comment]
> Is there a constraint pod status object for every instance of a constraint template?

generally, it's for every constraint object

File: internal/controller/constraint_controller.go (hunk 1 above)
Comment (zreigz) [reply internal/controller/constraint_controller.go]: done

**Comment by zreigz** [comment]
@michaeljguarino PTAL

**Review by michaeljguarino — APPROVED** [review]
This lgtm but looks like there's a go mod conflict

**Pull request merged**